apps/backend/src/filters/authentication-exception.filter.ts

```diff
@@ -1,24 +1,46 @@
 import {ArgumentsHost, Catch, ExceptionFilter} from '@nestjs/common';
 import * as _ from 'lodash';
+import winston from 'winston';
 import {ConfigService} from '../config/config.service';
 
 @Catch(Error)
 export class AuthenticationExceptionFilter implements ExceptionFilter {
   configService = new ConfigService();
+
+  private readonly line = '_______________________________________________\n';
+  public loggingTimeFormat = 'MMM-DD-YYYY HH:mm:ss Z';
+  public logger = winston.createLogger({
+    transports: [new winston.transports.Console()],
+    format: winston.format.combine(
+      winston.format.timestamp({
+        format: this.loggingTimeFormat
+      }),
+      winston.format.printf(
+        (info) =>
+          `${this.line}[${[info.timestamp]}] (Authentication Exception Filter): ${info.message}`
+      )
+    )
+  });
+
   catch(exception: Error, host: ArgumentsHost): void {
     const ctx = host.switchToHttp();
     const request = ctx.getRequest();
     const response = ctx.getResponse();
-    // Restart OIDC auth flow if state fails to verify, this is a known bug with passport-openidconnect that occours sometimes when logging in.
-    if (
-      _.get(request, 'authInfo.message') ===
-      'Unable to verify authorization request state.'
-    ) {
-      return response.redirect(301, '/authn/oidc');
-    }
-    response.cookie('authenticationError', exception.message, {
+    const errInfo = {
+      message: exception.message,
+      stack: exception.stack,
+      authInfo: _.get(request, 'authInfo'),
+      query: request.query,
+      headers: request.headers
+    };
+    this.logger.warn(
+      `Authentication Error\n${JSON.stringify(errInfo, null, 2)}`
+    );
+    const authError =
+      `${_.get(request, 'authInfo.message')}\n${exception.message}`.trim();
+    response.cookie('authenticationError', authError, {
       secure: this.configService.isInProductionMode()
     });
-    response.redirect(301, '/');
+    response.redirect(302, '/');
```

Comment:
Do you mind explaining the switch from 301 to 302, so now we are saying there is a temporary redirect? Just wanting to understand.

Reply:
It came up in discussion with the sponsor team how 302 made a bit more sense instead of 301 since you aren't permanently going to go to the '/' page based off of what could be a temporary auth problem.